java/core/src/main/java/com/lancedb/lance/DatasetFragment.java

```diff
@@ -33,23 +34,49 @@ public class DatasetFragment {
     this.metadata = metadata;
   }
 
+  /**
+   * Create a new Fragment Scanner.
+   *
+   * @param batchSize the scan options with batch size, columns filter, and substrait
+   * @return a fragment scanner
+   */
+  public Scanner newScan(long batchSize) {
+    return newScan(new ScanOptions(batchSize), Optional.empty());
+  }
+
+  /**
+   * Create a new Fragment Scanner.
+   *
+   * @param options the scan options
+   * @return a fragment scanner
+   */
+  public Scanner newScan(ScanOptions options) {
+    return newScan(options, Optional.empty());
+  }
+
+  /**
+   * Create a new Fragment Scanner.
+   *
+   * @param options the scan options with batch size, columns filter, and substrait
+   * @param filter the string filter e.g. id > 20
+   * @return a fragment scanner
+   */
+  public Scanner newScan(ScanOptions options, Optional<String> filter) {
+    return new FragmentScanner(dataset, metadata.getId(), options, filter, dataset.allocator);
+  }
+
   private native int countRowsNative(Dataset dataset, long fragmentId);
```

Comment:
we also have an countRow with filter API, should we leverage that as well?

Reply:
The Opened Scanner has Schema, Count Row API. One of my TODO is "Store the Scanner object in JNI, and get schema / count rows from the Scanner Object", is it fulfill the upstream requirement?